rust/query/mod.rs

```diff
@@ -20,6 +20,65 @@
  *
  */
 
-pub struct TypeQLMatch {
+use std::fmt;
+use std::fmt::{Display, Formatter};
 
-}
\ No newline at end of file
+use crate::pattern::*;
+use crate::{enum_getter, var};
+
+mod typeql_match;
+pub use typeql_match::*;
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum Query {
+    Match(TypeQLMatch),
+    Dud(()),
+}
+
+impl Query {
+    enum_getter!(into_match, Match, TypeQLMatch);
+
+    pub fn get<T: Into<String>, const N: usize>(self, vars: [T; N]) -> Query {
```

Comment:
`[T; N]` gains ownership of the array, so that we may move out of it into our structure values. `&[T]` will borrow, and we'll be forced to copy/clone. `[T]` is illegal, as arrays are on stack and must specify their size.